modules/acm/src/test/scala/edu/gemini/epics/acm/ObserveStateSpec.scala

```diff
@@ -0,0 +1,402 @@
+    (context.createChannel(_: String)).expects("gm:apply.DIR").returns(null)
+    (context.createChannel(_: String)).expects("gm:apply.VAL").returns(null)
+    (context.createChannel(_: String)).expects("gm:apply.MESS").returns(null)
+    (context.createChannel(_: String)).expects("gm:applyC.CLID").returns(null)
+    (context.createChannel(_: String)).expects("gm:applyC.VAL").returns(null)
+    (context.createChannel(_: String)).expects("gm:applyC.OMSS").returns(null)
+    val epicsService = new EpicsService(context)
+    val observe = new CaObserveSenderImpl(
+      "gmos::observeCmd",
+      "gm:apply",
+      "gm:applyC",
+      "gm:dc:observeC",
+      "gm:stop",
+      "gm:abort",
+      "GMOS Observe",
+      classOf[CarState],
+      epicsService)
+    // Start idle
+    assert(observe.applyState().isIdle())
+    // Post an observe
+    observe.post()
+
+    // CONFIGURATION
+    // APPLY Goes to busy
+    // VAL change
+    observe.onApplyValChange(4169)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4169)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+
+    // APPLY Goes to IDLE
+    // Another VAL change
+    observe.onApplyValChange(4169)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4169)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.IDLE)
+    assert(!observe.applyState().isIdle())
+
+    // OBSERVE
+    // OBSERVE goes BUSY
+    // VAL change
+    observe.onApplyValChange(4170)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4170)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+    // Another VAL change
+    observe.onApplyValChange(4170)
+    assert(!observe.applyState().isIdle())
+    // Observe CAR VAL change
+    observe.onObserveCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4170)
+    assert(!observe.applyState().isIdle())
+    // Apply goes IDLE
+    observe.onCarValChange(CarState.IDLE)
+    assert(!observe.applyState().isIdle())
+
+    // PAUSE observations
+    observe.onApplyValChange(4171)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4171)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+    // Another VAL change
+    observe.onApplyValChange(4171)
+    assert(!observe.applyState().isIdle())
+    // Observe CAR VAL change
+    observe.onObserveCarValChange(CarState.PAUSED)
+    // We are now idle
+    assert(observe.applyState().isIdle())
```

Comment:
@jluhrs Does this seem right? Should the apply go back to idle when paused?

Reply:
Yes, that is correct.